src/compressed_tensors/compressors/model_compressors/model_compressor.py

```diff
@@ -308,16 +307,28 @@ def decompress(self, model_path: str, model: Module):
         :param model: pytorch model to load decompressed weights into
         """
         model_path = get_safetensors_folder(model_path)
+        sparse_decompressed = False
+        if self.quantization_compressor is not None:
+            # update model structure
+            names_to_scheme = apply_quantization_config(model, self.quantization_config)
+            load_pretrained_quantization(model, model_path)
+
         if self.sparsity_compressor is not None:
+            # sparse decompression
             dense_gen = self.sparsity_compressor.decompress(model_path)
             self._replace_weights(dense_gen, model)
             setattr(model, SPARSITY_CONFIG_NAME, self.sparsity_compressor.config)
+            sparse_decompressed = True
 
         if self.quantization_compressor is not None:
+            # quantized decompression
             names_to_scheme = apply_quantization_config(model, self.quantization_config)
-            load_pretrained_quantization(model, model_path)
+            model_path_or_state_dict = (
+                model.state_dict() if sparse_decompressed else model_path
+            )
```

Comment:
why did we remove `load_pretrained_quantization`? what happens when there's only quantization?

Reply:
theres duplicate 
if self.quantization_compressor is not None:

Reply:
The duplicate condition above is needed to update the structure of the model for now, Will update in a follow up to completely isolate these two compressions. 

@dsikka `load_pretrained_quantization` this call is in the above if condition, it works for just quantized cases as well.

Reply:
Remove the unnecessary apply_quantization_config call